From ef75c0a52e5e3f4723d0142c31ea64f3ad8044d2 Mon Sep 17 00:00:00 2001
From: Alan Gates <gates@apache.org>
Date: Wed, 4 Feb 2015 19:17:48 +0000
Subject: [PATCH 0560/1164] CDH-39324 : HIVE-9550
 ObjectStore.getNextNotification() can return
 events inside NotificationEventResponse as null
 which conflicts with its thrift "required" tag

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1657377 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit eca51621f89614a4e3b0c7f0d4d7a504f6ede295)

Change-Id: Iee1c9321f621cfd790f59eee808729b1beb14ea5
---
 .../apache/hadoop/hive/metastore/ObjectStore.java  |    1 +
 1 file changed, 1 insertion(+)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
index 71a2519..ddbff0e 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
@@ -7496,6 +7496,7 @@ public NotificationEventResponse getNextNotification(NotificationEventRequest rq
       }
       Iterator<MNotificationLog> i = events.iterator();
       NotificationEventResponse result = new NotificationEventResponse();
+      result.setEvents(new ArrayList<NotificationEvent>());
       int maxEvents = rqst.getMaxEvents() > 0 ? rqst.getMaxEvents() : Integer.MAX_VALUE;
       int numEvents = 0;
       while (i.hasNext() && numEvents++ < maxEvents) {
-- 
1.7.9.5

